### ISSUE TITLE
Instructions for usage are not clear

### ISSUE THREAD

**Issue by ospfranco** [johanlantz/curly#9]
Hey trying to do a simple request using this library, but I'm not able to understand which parameters am I supposed to pass to it. I took a look into the test folder and the code there is not working for me. Here is my latest attempt:

```cpp
void onHttpRequestCompleted(curly_http_transaction_handle handle, long http_response_code, void* data, long size) {
  std::cout << "http request completed " << http_response_code << std::endl;
}

void CalendarHelper::fetchCalendar(std::string credentials) {
  std::cout << "Should fetch calendar in cpp" << std::endl;
  json j = json::parse(credentials);
//  std::cout << "access_token" << j["tokens"]["access_token"] << std::endl;
  std::cout << "fetching calendar" << std::endl;
  std::string accessToken = std::string(j["tokens"]["access_token"]);
  std::string urlStr = "https://www.googleapis.com/calendar/v3/calendars/primary/events?access_token=" + accessToken + "&maxResults=100&singleEvents-trye";
  const char* headers = "{\"Accept\":\"application/json\"}";
  curly_http_transaction_handle handle = curly_http_get(urlStr.c_str(), headers, &onHttpRequestCompleted); // No matching function call to 'curly_http_get'
}

```

**Comment by johanlantz** [comment]
Happy you found this little project. However, this is something I made a very long time ago so it is not really maintained anymore. I know curl can be a bit cumbersome to use but I would try using it directly since there is a good mailing list. There might also be similar project to mine that wraps a simpler API on top of curl, I think I have encountered something like that in the past.

If you want to pursue using curly, then feel free to open a PR with a better example.
